### PR TITLE
Clean up AddressbookGuiTest #459

### PR DESCRIPTION
**Pull request by yamgent**
Part of #459.

Proposed merge commit message:
```
Let's clean up AddressBookGuiTest by:

- Removing the while loop that waits for the stage to show in the 
  setup() phase. FxToolkit#showStage() automatically blocks until
  the stage shows, so the wait is unnecessary.
- Creating the guiRobot only in AddressBookGuiTest. It need not be
  recreated and can be shared by everyone.
- Reducing the visibility of assertMatching(...) and raise(...) from 
  public to private.
- Renaming method parameters that does not conform to coding standards.
- Renaming setupSpec() to setupOnce().
- Changing cleanup()'s throws declaration from TimeoutException to 
  Exception.
- Renaming assertMatching(...) to assertCardMatchesPerson(...).
```

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@yamgent submitted v1 for review.

* [v1 1/3] [AddressBookGuiTest\#setup\(\): Remove redundant wait for stage to show](https://github.com/se-edu/addressbook-level4/pull/520/commits/c83c665bcfad38542822d8f91fce9b60a281c8cd)
* [v1 2/3] [AddressBookGuiTest\#setup\(\): Remove redundant EventsCenter reset](https://github.com/se-edu/addressbook-level4/pull/520/commits/732fff0f7534235b6ed73ee600cb8f56b3fe654f)
* [v1 3/3] [AddressBookGuiTest: Add guiRobot](https://github.com/se-edu/addressbook-level4/pull/520/commits/c8d836117772b7312a091c58979792d14f547748)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/d4dd4c253b3218faf475d5d3941e5db377e62637...c8d836117772b7312a091c58979792d14f547748))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/520/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/guitests/AddressBookGuiTest.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public void setup() throws Exception {
             statusBarFooter = mainGui.getStatusBarFooter();
             this.stage = stage;
         });
-        EventsCenter.clearSubscribers();
```
Comment (yamgent) [inline src/test/java/guitests/AddressBookGuiTest.java]: I recall we discussed that for instance variables, they are reset for each test cases. However, that is not true for class (i.e. `static`) variables.

`EventsCenter` is a singleton here implemented using `static`. Perhaps this commit is wrong and just happens to work by chance.

File: src/test/java/guitests/AddressBookGuiTest.java [hunk 2]
```diff
@@ -40,6 +40,7 @@
     public TestName name = new TestName();
 
     protected TypicalPersons td = new TypicalPersons();
+    protected GuiRobot guiRobot = new GuiRobot();
```
Comment (yamgent) [inline src/test/java/guitests/AddressBookGuiTest.java]: Benefit may not show clearly until PR #512 is merged. Will put this PR on hold and wait for #512.

File: src/test/java/guitests/AddressBookGuiTest.java (hunk 1 above)
Comment (Zhiyuan-Amos) [inline src/test/java/guitests/AddressBookGuiTest.java]: Hmm I see. That means we should delete this commit?

**Comment by CanIHasReview** [comment]
# v2

@yamgent submitted v2 for review.

* [v2 1/5] [AddressBookGuiTest\#setup\(\): Remove redundant wait for stage to show](https://github.com/se-edu/addressbook-level4/pull/520/commits/dc43834458e38fe011dbff0081aa09e99f282d6a)
* [v2 2/5] [AddressBookGuiTest: Add guiRobot](https://github.com/se-edu/addressbook-level4/pull/520/commits/679de5d6f488e76035d20dee69b6478904ba2621)
* [v2 3/5] [AddressBookGuiTest\#raise\(BaseEvent\): Rename parameter from e to event](https://github.com/se-edu/addressbook-level4/pull/520/commits/9987ec0aadd7eafeb2c80f1210fcce63384e0baf)
* [v2 4/5] [AddressBookGuiTest\#setupSpec\(\): Rename to setupOnce\(\)](https://github.com/se-edu/addressbook-level4/pull/520/commits/3c91ad3a6d63c57d2ac537cc4622e05917c0cb6d)
* [v2 5/5] [AddressBookGuiTest: Make assertMatching\(\.\.\.\) and raise\(\.\.\.\) protected](https://github.com/se-edu/addressbook-level4/pull/520/commits/bd39c0ae91bb72860a00fd8a77b36e8832017377)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/daddc4289462285c4474311539b79287465273cc...bd39c0ae91bb72860a00fd8a77b36e8832017377)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/c563b031c8902b69e656eb0497a695ef6cbde688~1...c563b031c8902b69e656eb0497a695ef6cbde688))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/520/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/guitests/AddressBookGuiTest.java [hunk 3]
```diff
@@ -109,7 +109,7 @@ public void cleanup() throws TimeoutException {
     /**
      * Asserts the person shown in the card is same as the given person
      */
-    public void assertMatching(ReadOnlyPerson person, PersonCardHandle card) {
+    protected void assertMatching(ReadOnlyPerson person, PersonCardHandle card) {
```
Comment (Zhiyuan-Amos) [inline src/test/java/guitests/AddressBookGuiTest.java]: method name can be more descriptive, something like `personAndCardMatch`? Also I'm thinking whether we should change the method signature to be `(PersonCardHandle, ReadOnlyPerson)` to follow the method signature of `TestUtil#compareCardAndPerson`

**Comment by CanIHasReview** [comment]
# v3

@yamgent submitted v3 for review.

* [v3 1/7] [AddressBookGuiTest\#setup\(\): Remove redundant wait for stage to show](https://github.com/se-edu/addressbook-level4/pull/520/commits/dc43834458e38fe011dbff0081aa09e99f282d6a)
* [v3 2/7] [AddressBookGuiTest: Add guiRobot](https://github.com/se-edu/addressbook-level4/pull/520/commits/679de5d6f488e76035d20dee69b6478904ba2621)
* [v3 3/7] [AddressBookGuiTest\#raise\(BaseEvent\): Rename parameter from e to event](https://github.com/se-edu/addressbook-level4/pull/520/commits/9987ec0aadd7eafeb2c80f1210fcce63384e0baf)
* [v3 4/7] [AddressBookGuiTest\#setupSpec\(\): Rename to setupOnce\(\)](https://github.com/se-edu/addressbook-level4/pull/520/commits/3c91ad3a6d63c57d2ac537cc4622e05917c0cb6d)
* [v3 5/7] [AddressBookGuiTest: Make assertMatching\(\.\.\.\) and raise\(\.\.\.\) protected](https://github.com/se-edu/addressbook-level4/pull/520/commits/bd39c0ae91bb72860a00fd8a77b36e8832017377)
* [v3 6/7] [AddressBookGuiTest\#cleanup\(\): Change throws declaration to Exception](https://github.com/se-edu/addressbook-level4/pull/520/commits/0e3422e7949ccb1d54ed5cdf85385dfd5a573d0e)
* [v3 7/7] [AddressBookGuiTest\#assertMatching\(\.\.\.\): Rename method](https://github.com/se-edu/addressbook-level4/pull/520/commits/9c5882a29c73e0e6f332dec386c265bb4b59153f)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/daddc4289462285c4474311539b79287465273cc...9c5882a29c73e0e6f332dec386c265bb4b59153f)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/4d108641618bd1fb23e63b153256c502cb8beb3f~1...4d108641618bd1fb23e63b153256c502cb8beb3f))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/520/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Pull request merged**